packages/compiler/core/checker.ts

```diff
@@ -4674,33 +4671,54 @@ export function filterModelProperties(
   return finishTypeForProgram(program, newModel);
 }
 
+/**
+ * Gets the property from the nearest base type that is overridden by the
+ * given property, if any.
+ */
+export function getOverriddenProperty(property: ModelProperty): ModelProperty | undefined {
+  compilerAssert(
+    property.model,
+    "Parent model must be set before overridden property can be found."
+  );
+
+  for (let current = property.model.baseModel; current; current = current.baseModel) {
+    const overriden = current.properties.get(property.name);
+    if (overriden) {
+      return overriden;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * Enumerates the properties declared by model or inherited from its base.
+ *
+ * Properties declared by more derived types are enumerated before properties
+ * of less derived types.
+ *
+ * Properties that are overridden are not enumerated.
```

Comment:
Adding documentation for walkInheritedProperties to cover the ambiguity around what it does in the presence of overrides now.